cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionHandler.java

```diff
@@ -135,4 +164,153 @@ public void deleteConnection(HttpServiceRequest request, HttpServiceResponder re
+   */
+  @GET
+  @Path(API_VERSION + "/contexts/{context}/connections/{connection}/browse")
+  public void browse(HttpServiceRequest request, HttpServiceResponder responder,
+                     @PathParam("context") String namespace,
+                     @PathParam("connection") String connection,
+                     @QueryParam("path") String path,
+                     @QueryParam("limit") @DefaultValue("1000") int limit,
+                     @QueryParam("browseAll") @DefaultValue("false") boolean browseAll) {
+    respond(namespace, responder, namespaceSummary -> {
+      if (namespaceSummary.getName().equalsIgnoreCase(NamespaceId.SYSTEM.getNamespace())) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST,
+                            "Browsing connection in system namespace is currently not supported");
+        return;
+      }
+
+      if (path == null) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST, "Path is not provided in the browse request");
+        return;
+      }
+
+      if (!browseAll && limit <= 0) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST, "Limit should be greater than 0.");
+        return;
+      }
+
+      PluginConfigurer pluginConfigurer = getContext().createPluginConfigurer(namespaceSummary.getName());
+      Connection conn = store.getConnection(new ConnectionId(namespaceSummary, connection));
+
+      try (Connector connector = getConnector(pluginConfigurer, conn.getPlugin())) {
+        connector.configure(pluginConfigurer);
+        BrowseRequest browseRequest = BrowseRequest.builder(path).setLimit(browseAll ? null : limit).build();
+        responder.sendJson(connector.browse(browseRequest));
+      }
+    });
+  }
+
+  /**
+   * Retrive sample result for the connection
+   */
+  @POST
+  @Path(API_VERSION + "/contexts/{context}/connections/{connection}/sample")
+  public void sample(HttpServiceRequest request, HttpServiceResponder responder,
+                     @PathParam("context") String namespace,
+                     @PathParam("connection") String connection,
+                     @QueryParam("path") String path,
+                     @QueryParam("limit") @DefaultValue("1000") int limit) {
+    respond(namespace, responder, namespaceSummary -> {
+      if (namespaceSummary.getName().equalsIgnoreCase(NamespaceId.SYSTEM.getNamespace())) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST,
+                            "Sampling connection in system namespace is currently not supported");
+        return;
+      }
+
+      if (path == null) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST, "Path is not provided in the sample request");
+        return;
+      }
+
+      if (limit <= 0) {
+        responder.sendError(HttpURLConnection.HTTP_BAD_REQUEST, "Limit should be greater than 0");
+        return;
+      }
+
+      String propertyString = StandardCharsets.UTF_8.decode(request.getContent()).toString();
```

Comment:
Even if there is just one properties field right now, it's better for the request body to be an object instead of just the properties map, as it allows us to add things to the request body in the future.

Reply:
Also, if we have a request body, all the request information should be in the body instead of some things in the body and some as query params. Would be better to make the SampleRequest itself the request body.